### PR TITLE
Add account media prune and fetch command

### PR DESCRIPTION
**Pull request by dunkelstern**
This implements two new sub-commands for `tootctl media`:

- `prune-profile-media` to remove old avatar and header images
- `fetch-profile-media` to download missing avatar and header images basically undoing the prune command

I am new to Ruby and ActiveRecord, so please advise if something is wrong, I will fix the errors if needed.

This is a fix to #9567

### CODE REVIEW

File: lib/mastodon/media_cli.rb
Comment (connorshea) [inline lib/mastodon/media_cli.rb:76]: Normal convention in ruby would be to use `!` and `&&` instead of `not` and `and` (I've been a Ruby dev for 7 years and I actually thought `not` here was a syntax error, I didn't even realize it was valid syntax until I googled it lol).

```suggestion
        next if (!account.avatar.present? && !account.header.present?)
        next if !aggressive && Follow.where(account: account).or(Follow.where(target_account: account)).count > 0
```

File: lib/mastodon/media_cli.rb
Comment (connorshea) [inline lib/mastodon/media_cli.rb:112]: Same thing with `or`

```suggestion
          unless account.avatar.present? || account.avatar_remote_url.blank?
```

File: lib/mastodon/media_cli.rb
Comment (connorshea) [inline lib/mastodon/media_cli.rb:118]:
```suggestion
          unless account.header.present? || account.header_remote_url.blank?
```

**Review by connorshea — COMMENTED** [review]
Just syntax nits, otherwise this looks more-or-less fine to me.

**Comment by rgrunbla** [comment]
The basis for deletion by this script is the datetime of the last webfinger request : could you provide examples of things that would result in webfinger requests ? E.g. if the remote account is already known by the local instance, and the remote account does not change profile picture / header / note, do you still have regular webfinger request to the account ?

File: lib/mastodon/media_cli.rb
Comment (dunkelstern) [reply lib/mastodon/media_cli.rb:76]: Ok that's probably the python speaking 🙂 will fix shortly

**Comment by dunkelstern** [comment]
As I understand the `last_webfingered_at` date resolves mainly to the `Account.possibly_stale?` property. This marks an account as stale if the last webfinger is older than 1 day (or `nil`). So if the account is used by something incoming i think the webfinger process is restarted after about one day.

I think (if i understand correctly) mastodon will trigger a re-download of a missing avatar or header image when it processes an activity including that user. `ActivityPub::Activity::Update` calls `ActivityPub::ProcessAccountService` in its `update_account` function when the ActivityPub message contains an actor. `ProcessAccountService` will then run `RedownloadAvatarWorker` which checks for an empty avatar and if a remote url is set triggers a download of the remote asset file.

**Review by connorshea — APPROVED** [review]
LGTM 👍

**Comment by dunkelstern** [comment]
Ok i see some linter stuff. Will fix this evening. Seems the linter in mu vscode setup did not run.

**Review by ineffyble — CHANGES_REQUESTED** [review]
Can you fix the linter issue? or alternatively wait until #21477 is merged, at which point the linter will be happy.

**Comment by dunkelstern** [comment]
I am afraid I cannot fix all linter problems as the linter complains about complexity of a function that was already there:

```
Perceived complexity for remove_orphans is too high. [26/25]
```

**Comment by dunkelstern** [comment]
Ok, I have to retract this, the webfinger is not updated as often as I suspected. I'll have to find another way of doing this. `updated_at` also doesn't work. Perhaps the last posted status we have seen? I think I know why this has never been implemented. It's really non-trivial and full of edge cases :(

I'll experiment a bit more on it and will retry in a few days with a new PR. Sorry for the work I produced for my fellow reviewers.

**Pull request closed without merging**
